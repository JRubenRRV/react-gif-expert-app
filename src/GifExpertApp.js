import React, { useState } from 'react'

export const GifExpertApp = () => {

    // const categories = ['Redbull', 'Merceds', 'Ferrari', 'Mclaren'];
    const [categories, setCategories] = useState(['Redbull', 'Merceds', 'Ferrari', 'Mclaren']);

    const handleAdd = () => {
        // setCategories([...categories, 'Alpine']);
        setCategories(cats => [...categories, 'Alpine']);
    }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr/>
            <button onClick={handleAdd}>Agregar</button>
            <ol>
                {
                    categories.map(category => {
                        return <li key={category}>{category}</li>;
                    })
                }
            </ol>
        </>
    )
}
