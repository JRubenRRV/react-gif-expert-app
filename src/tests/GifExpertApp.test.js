import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas a GifExpertApp', () => {

    test('debe renderizarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar una lista de categorias', () => {
        const categories = ['lego', 'spartan'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
})