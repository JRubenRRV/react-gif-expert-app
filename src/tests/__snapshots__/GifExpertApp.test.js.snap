// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="lego"
      key="lego"
    />
    <GifGrid
      category="spartan"
      key="spartan"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas a GifExpertApp debe mostrar una lista de categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="lego"
      key="lego"
    />
    <GifGrid
      category="spartan"
      key="spartan"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas a GifExpertApp debe renderizarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
